packages/@aws-cdk/aws-apigatewayv2/lib/api.ts

```diff
@@ -0,0 +1,186 @@
+   * @default - ID of the HttpApi construct.
+   */
+  readonly apiName?: string;
+  /**
+   * API protocol
+   *
+   * @default HTTP
+   */
+  readonly protocol?: ProtocolType;
+  /**
+   * target lambda function for lambda proxy integration.
+   *
+   * @default - None. Specify either `targetHandler` or `targetUrl`
+   */
+  readonly targetHandler?: lambda.IFunction;
+
+  /**
+   * target URL for HTTP proxy integration.
+   *
+   * @default - None. Specify either `targetHandler` or `targetUrl`
+   */
+  readonly targetUrl?: string;
+}
+
+/**
+ * protocol types for the Amazon API Gateway HTTP API
+ */
+export enum ProtocolType {
+  /**
+   * HTTP protocol
+   */
+  HTTP = 'HTTP',
+  /**
+   * Websocket protocol
+   */
+  WEBSOCKET = 'WEBSOCKET'
+}
+
+/**
+ * HTTPApi Resource Class
+ *
+ * @resource AWS::ApiGatewayV2::Api
+ */
+export class HttpApi extends ApiBase implements IHttpApi {
+  /**
+   * import from ApiId
+   */
+  public static fromApiId(scope: cdk.Construct, id: string, httpApiId: string): IHttpApi {
+    class Import extends cdk.Resource implements IHttpApi {
+      public readonly httpApiId = httpApiId;
+    }
+    return new Import(scope, id);
+  }
+  /**
+   * the API identifer
+   */
+  public readonly httpApiId: string;
+  /**
+   * AWS partition either `aws` or `aws-cn`
+   */
+  public readonly partition: string;
+  /**
+   * AWS region of this stack
+   */
+  public readonly region: string;
+  /**
+   * AWS account of this stack
+   */
+  public readonly account: string;
+  /**
+   * AWS domain name either `amazonaws.com` or `amazonaws.com.cn`
+   */
+  public readonly awsdn: string;
+  /**
+   * the full URL of this API
+   */
+  public readonly url: string;
+  /**
+   * root route
+   */
+  public root?: IRouteBase;
+
+  constructor(scope: cdk.Construct, id: string, props?: HttpApiProps) {
+    super(scope, id, {
+      physicalName: props?.apiName || id,
+    });
+
+    // if ((!props) ||
+    // (props!.targetHandler && props!.targetUrl) ||
+    // (props!.targetHandler === undefined && props!.targetUrl === undefined)) {
+    //   throw new Error('You must specify either a targetHandler or targetUrl, use at most one');
+    // }
+
+    if (props?.targetHandler && props.targetUrl) {
+      throw new Error('You must specify either a targetHandler or targetUrl, use at most one');
+    }
+
+    this.region = cdk.Stack.of(this).region;
+    this.partition = this.isChina() ? 'aws-cn' : 'aws';
+    this.account = cdk.Stack.of(this).account;
+    this.awsdn = this.isChina() ? 'amazonaws.com.cn' : 'amazonaws.com';
+
+    const apiProps: CfnApiProps = {
+      name: this.physicalName,
+      protocolType: props?.protocol ?? ProtocolType.HTTP,
+      target: props?.targetHandler ? props.targetHandler.functionArn : props?.targetUrl ?? undefined
+    };
+    const api = new CfnApi(this, 'Resource', apiProps );
+    this.httpApiId = api.ref;
+
+    this.url = `https://${this.httpApiId}.execute-api.${this.region}.${this.awsdn}`;
+
```

Comment:
You shouldn't be doing any of this.

Take a look at [how this is done in `RestApi`](https://github.com/aws/aws-cdk/blob/5dc1407d78e63d7690cf5d171aec92f4191f0715/packages/%40aws-cdk/aws-apigateway/lib/restapi.ts#L289) and use the same pattern.